### PR TITLE
add opentelemetry

### PR DESCRIPTION
**Pull request by Zach41**
add RPC tracing with opentelemetry in milvus
ISSUE: #18873 
Signed-off-by: Zach41 <zongmei.zhang@zilliz.com>

### CODE REVIEW

**Comment by Zach41** [comment]
example for proxy/search
image
image

**Comment by Zach41** [comment]
rerun ut

**Comment by Zach41** [comment]
rerun ut

**Comment by Zach41** [comment]
rerun ut

**Comment by Zach41** [comment]
rerun ut

**Comment by Zach41** [comment]
rerun ut

File: internal/util/trace/span.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+func (s *Span) RecordBool(key string, val bool) {
+	s.SetAttributes(attribute.Bool(key, val))
+}
+
+func (s *Span) RecordAny(key string, val any) {
+	payload, err := json.Marshal(val)
+	if err != nil {
+		s.SetAttributes(attribute.String(key, "{}"))
+	} else {
```
Comment (congqixia) [inline internal/util/trace/span.go:65]: Need some method to detect serialization failure other than swallow this error.

File: internal/util/trace/util.go [hunk 2]
```diff
@@ -13,208 +13,149 @@ package trace
...
+	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/zap"
 )
 
+const (
+	sampleOutputKey   = "opentelemetry.sampleFile"
+	jaegerEndpointKey = "opentelemetry.jaegerEndpoint"
+	sampleFracKey     = "opentelemetry.sampleFraction"
+	devKey            = "development"
+)
```
Comment (congqixia) [inline internal/util/trace/util.go:37]: Config parsing logic shall be defined in params type.

**Review by congqixia — COMMENTED** [review]
span tags format are not unified.
Found snake based, dot connected tags, shall we put these tags in some common place?

**Comment by Zach41** [comment]
> span tags format are not unified. Found snake based, dot connected tags, shall we put these tags in some common place?

tag keys are simply user defined string, it's hard to unify.

**Pull request closed without merging**